woocommerce/woocommerce
# [Accessibility] Make the edit/delete actions of shipping/products accessible in manual orders

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #26875 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Orders -> Add Order.
2. Click on the "Add item(s)" button.
3. Click on the "Add shipping" button.
4. Tab navigate through the page and verify that you can focus on both the "Edit shipping" and "Delete shipping" buttons.
![image](https://github.com/woocommerce/woocommerce/assets/9420947/0101df73-7cca-42ef-80ed-f1ac34f78a69)

5. Use a screen reader and verify that "Edit shipping" and "Delete shipping" are announced once the correspondent buttons are focused.
6. Hover over the Shipping line and verify that the "Edit shipping" and "Delete shipping" show up.
7. Hover off the Shipping line and verify that the "Edit shipping" and "Delete shipping" disappear.
8. Add a product by clicking on the "Add product(s)" buttons and repeat the steps above but this time using the "Edit item" and "Delete item" buttons on the Product line.
![image](https://github.com/woocommerce/woocommerce/assets/9420947/73ae5d68-6afb-436a-aa2f-13fdcff7e6a0)
9. Add a fee by clicking on the "Add fee" button and repeat the steps above but this time using the "Edit fee" and "Delete fee" buttons on the Fee line.
![image](https://github.com/woocommerce/woocommerce/assets/9420947/f7d49d39-8a51-465f-aa1d-aa68c916f955)





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
